src/js/collections/DataPackage.js

```diff
@@ -415,6 +415,115 @@ define([
         }
       },
 
+      /**
+       * Fetches member models in batches to avoid fetching all members simultaneously.
+       *
+       * @param {Array} models - The array of member models to fetch.
+       * @param {number} [index=0] - The current index of the model being fetched.
+       * @param {number} [batchSize=10] - The number of models to fetch in each batch.
+       * @param {number} [timeout=5000] - The timeout for each fetch request in milliseconds.
+       * @param {number} [maxRetries=3] - The maximum number of retries for each fetch request.
+       */
+      fetchMemberModels(models, index = 0, batchSize = 10, timeout = 5000, maxRetries = 3) {
+        // Update the number of file metadata items being loaded
+        this.packageModel.set("numLoadingFileMetadata", models.length - index);
+
+        // If the index is greater than or equal to the length of the models array, stop fetching
+        if (index >= models.length) {
```

Comment:
```suggestion
        if (index >== models.length) {
```

just an example of a [linting rule](https://eslint.org/docs/latest/rules/eqeqeq) that you can use eslint detect.